engine/engine/src/test/wscript

```diff
@@ -1,9 +1,31 @@
 #! /usr/bin/env python
 
-import waflib.Task, waflib.TaskGen, waflib.Options
+import waflib.Task, waflib.TaskGen
+from waflib import Options
 import os, sys, glob, re
+import run, sdk
 import waf_gamesys
 
+# 1. We want to run it with run.py so that it logs nicely with [exec] ...
+# 2. We need to add DYLD_INSERT_LIBRARIES for macOS
+def run_bob(task):
+    env = dict(os.environ)
+    merged_table = task.env.get_merged_dict()
+    keys=list(merged_table.keys())
+    for key in keys:
+        v = merged_table[key]
+        if isinstance(v, str):
+            env[key] = v
+
+    if Options.options.with_asan:
+        platform = env['PLATFORM']
+        if platform in ('x86_64-macos', 'arm64-macos', 'arm64-ios', 'x86_64-ios'):
+            sdkinfo = sdk.get_sdk_info(sdk.SDK_ROOT, platform)
+            env['DYLD_INSERT_LIBRARIES'] = sdkinfo['asan']['path']
```

Comment:
The issue was that I hadn't added the 'skip_asan' feature of the new `modelc_shared` library.
And as such bob failed, since the shared library then couldn't find this asan library.

The easy fix would have been to add the `skip_asan` feature to the modelc_shared, but I've also wanted to cleanup my old technical debt which I added when adding the ASAN support.
In short, I think I should be able to remove the need for the `skip_asan` feature altogether. (But I'll do that in a future PR)